app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -0,0 +1,208 @@
+"use strict";
+(function() {
+  angular.module("openshiftConsole").component('editEnvironmentFrom', {
+    controller: [
+      '$attrs',
+      '$filter',
+      '$routeParams',
+      'keyValueEditorUtils',
+      EditEnvironmentFrom
+    ],
+    bindings: {
+      addRowLink: '@',              // creates a link to "add row" and sets its text label
+      entries: '=',                 // an array of objects containing configmaps and secrets
+      envFromSelectorOptions: '<',  // dropdown selector options, an array of objects
+      selectorPlaceholder: '@'      // placeholder copy for dropdown selector
+    },
+    templateUrl: 'views/directives/edit-environment-from.html'
+  });
+
+  function EditEnvironmentFrom($attrs,
+                               $filter,
+                               $routeParams,
+                               utils) {
+    var ctrl = this;
+
+    var canI = $filter('canI');
+    var humanizeKind = $filter('humanizeKind');
+    var namespace = $routeParams.project;
```

Comment:
Better to pass the project in than read it from route params. But why is it needed?

Reply:
For the comment chain, related to the below dummy values referenced in ```editEnvironmentFrom.js```